src/opentimelineio/effect.h

```diff
@@ -18,7 +18,7 @@ class Effect : public SerializableObjectWithMetadata {
            std::string const& effect_name = std::string(),
            AnyDictionary const& metadata = AnyDictionary());
 
-    std::string const& effect_name() const {
+    std::string const& effect_name() const noexcept {
         return _effect_name;
```

Comment:
```suggestion
    std::string effect_name() const noexcept {
        return _effect_name;
```
const ref in a return makes no guarantees that the string actually lives as long as the reference, so returning const& in general is an anti-pattern.